x-pack/metricbeat/module/azure/billing/client.go

```diff
@@ -46,33 +42,51 @@ func NewClient(config azure.Config) (*Client, error) {
 }
 
 // GetMetrics returns the usage detail and forecast values.
-func (client *Client) GetMetrics() (Usage, error) {
-
+func (client *Client) GetMetrics(startTime time.Time, endTime time.Time) (Usage, error) {
 	var usage Usage
 	scope := fmt.Sprintf("subscriptions/%s", client.Config.SubscriptionId)
 	if client.Config.BillingScopeDepartment != "" {
 		scope = fmt.Sprintf("/providers/Microsoft.Billing/departments/%s", client.Config.BillingScopeDepartment)
 	} else if client.Config.BillingScopeAccountId != "" {
 		scope = fmt.Sprintf("/providers/Microsoft.Billing/billingAccounts/%s", client.Config.BillingScopeAccountId)
 	}
-	startTime := time.Now().UTC().Truncate(24 * time.Hour).Add((-24) * time.Hour)
-	endTime := startTime.Add(time.Hour * 24).Add(time.Second * (-1))
-	usageDetails, err := client.BillingService.GetUsageDetails(scope, "properties/meterDetails",
-		fmt.Sprintf("properties/usageStart eq '%s' and properties/usageEnd eq '%s'", startTime.Format(time.RFC3339Nano), endTime.Format(time.RFC3339Nano)),
-		"", nil, "properties/instanceLocation")
+	client.Log.Infof("Getting usage details for scope: %s, startDate: %s, endDate: %s", scope, startTime, endTime)
```

Comment:
It does. It turns out the [logp](https://github.com/elastic/elastic-agent-libs/tree/main/logp) package is using zap under the hood, so we are good to go:

Here's an example:

```json
[
    {
        "log.level": "info",
        "@timestamp": "2022-06-24T08:03:53.752+0200",
        "log.logger": "azure billing",
        "log.origin": {
            "file.name": "billing/billing.go",
            "file.line": 69
        },
        "message": "Fetching billing data",
        "service.name": "metricbeat",
        "billing.start_time": "2022-06-23T00:00:00.000Z",
        "billing.end_time": "2022-06-23T23:59:59.000Z",
        "ecs.version": "1.6.0"
    },
    {
        "log.level": "info",
        "@timestamp": "2022-06-24T08:03:53.753+0200",
        "log.logger": "azure billing client",
        "log.origin": {
            "file.name": "billing/client.go",
            "file.line": 57
        },
        "message": "Getting usage details for scope",
        "service.name": "metricbeat",
        "billing.scope": "subscriptions/33ebbfb4-db72-4627-b106-f2309739d51f",
        "billing.start_time": "2022-06-23T00:00:00.000Z",
        "billing.end_time": "2022-06-23T23:59:59.000Z",
        "ecs.version": "1.6.0"
    },
    {
        "log.level": "warn",
        "@timestamp": "2022-06-24T08:04:00.131+0200",
        "log.logger": "azure billing service",
        "log.origin": {
            "file.name": "billing/service.go",
            "file.line": 82
        },
        "message": "no forecasts available for subscription; possibly because the subscription is not an enterprise subscription. For details, see: https://docs.microsoft.com/en-us/rest/api/consumption/",
        "service.name": "metricbeat",
        "billing.filter": "properties/chargeType eq 'Actual'",
        "billing.subscription_id": "33ebbfb4-db72-4627-b106-f2309739d51f",
        "ecs.version": "1.6.0"
    }
]
```